app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,12 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = MerchantsFacade.all_merchants
+
+  end
+
+  def show
+    @merchant = MerchantsFacade.merchant_details(params[:id])
+    @items = MerchantsFacade.merchants_items(params[:id])
+  end
```

Comment:
Nice job here. You could also use a MerchantItemsFacade, and then create MerchantItem objects to send back to the view.